src/main/kotlin/dev/gorokhov/smoothcaret/SmoothCaretRenderer.kt

```diff
@@ -131,16 +138,18 @@ class SmoothCaretRenderer(private val settings: SmoothCaretSettings) : CustomHig
                     settings.caretHeightMargins
                 }
 
-                when (settings.caretStyle) {
-                    SmoothCaretSettings.CaretStyle.BLOCK -> {
+                when (caret.visualAttributes.shape) {
+                    CaretVisualAttributes.Shape.BAR,
+                    CaretVisualAttributes.Shape.DEFAULT -> {
                         g2d.fillRect(caretX, caretY + yOffset, settings.caretWidth, scaledHeight)
                     }
-
-                    SmoothCaretSettings.CaretStyle.LINE -> {
-                        g2d.fillRect(caretX, caretY + yOffset, settings.caretWidth, scaledHeight)
+                    CaretVisualAttributes.Shape.BOX,
+                    CaretVisualAttributes.Shape.BLOCK -> {
+                        val nextChar = getNextCharFromCursorOrDefault(editor, caret, 'm')
+                        updateNextCharWidth(nextChar, editor)
+                        g2d.fillRect(caretX, caretY + yOffset, nextCharWidth,scaledHeight)
                     }
-
-                    SmoothCaretSettings.CaretStyle.UNDERSCORE -> {
+                    CaretVisualAttributes.Shape.UNDERSCORE -> {
```

Comment:
Note: I did 0 testing of the underscore state, IDK how to get to that caret shape, and it seems there's already something attempting to do an underscore of some sort (looks like it'd be quite a short underscore tho).

If someone knows how to get the underscore caret to show up, I'd be happy to test it (and, it probably needs to use my block width determination "algorithm" to get an underscore that spans the whole letter).